File: src/com/alchitry/labs/parsers/types/SyncLogic.java
package com.alchitry.labs.parsers.types;

import com.alchitry.labs.parsers.ConstValue;
import com.alchitry.labs.parsers.lucid.parser.LucidParser.ExprContext;

public interface SyncLogic {
	public String getName();

	public ExprContext getClk();

	public ExprContext getRst();

	public ConstValue getInit();
}

File: src/com/alchitry/labs/parsers/lucidv2/signals/SyncLogic.kt
package com.alchitry.labs.parsers.lucidv2.signals

import com.alchitry.labs.parsers.lucid.parser.LucidParser.ExprContext
import com.alchitry.labs.parsers.lucidv2.values.Value

interface SyncLogic {
    val init: Value
    val clkCtx: ExprContext
    val rstCtx: ExprContext?
    var value: Value
}